src/server/pkg/storage/chunk/chunk_test.go

```diff
@@ -3,66 +3,92 @@ package chunk
 import (
 	"bytes"
 	"context"
+	"fmt"
 	"io"
+	"math/rand"
+	"strconv"
 	"testing"
+	"time"
 
 	"github.com/chmduquesne/rollinghash/buzhash64"
 	"github.com/pachyderm/pachyderm/src/client/pkg/require"
+	"modernc.org/mathutil"
 )
 
 const (
 	averageBits = 23
 )
 
-func Write(t *testing.T, chunks *Storage, n, rangeSize int) ([]*DataRef, []byte) {
-	var finalDataRefs []*DataRef
-	var seq []byte
-	t.Run("Write", func(t *testing.T) {
-		f := func(_ *DataRef, annotations []*Annotation) error {
-			for _, a := range annotations {
-				finalDataRefs = append(finalDataRefs, a.NextDataRef)
-			}
-			return nil
-		}
-		w := chunks.NewWriter(context.Background(), averageBits, f, 0)
-		seq = RandSeq(n * MB)
-		for i := 0; i < n/rangeSize; i++ {
-			w.Annotate(&Annotation{
-				NextDataRef: &DataRef{},
-			})
-			_, err := w.Write(seq[i*MB*rangeSize : (i+1)*MB*rangeSize])
-			require.NoError(t, err)
-		}
-		require.NoError(t, w.Close())
-	})
-	return finalDataRefs, seq
+var tests = []struct {
+	maxAnnotationSize int
+	maxTagSize        int
+	n                 int
+}{
+	{1 * KB, 1 * KB, 1 * KB},
+	{1 * KB, 1 * KB, 1 * MB},
+	{1 * MB, 1 * KB, 100 * MB},
+	{1 * MB, 1 * MB, 100 * MB},
+	{10 * MB, 1 * MB, 100 * MB},
 }
 
 func TestWriteThenRead(t *testing.T) {
 	objC, chunks := LocalStorage(t)
 	defer Cleanup(objC, chunks)
-	finalDataRefs, seq := Write(t, chunks, 100, 1)
-	mid := len(finalDataRefs) / 2
-	initialRefs := finalDataRefs[:mid]
-	streamRefs := finalDataRefs[mid:]
-	r := chunks.NewReader(context.Background())
-	r.NextRange(initialRefs)
-	buf := &bytes.Buffer{}
-	t.Run("ReadInitial", func(t *testing.T) {
-		_, err := io.Copy(buf, r)
-		require.NoError(t, err)
-		require.Equal(t, bytes.Compare(buf.Bytes(), seq[:buf.Len()]), 0)
-	})
-	seq = seq[buf.Len():]
-	buf.Reset()
-	t.Run("ReadStream", func(t *testing.T) {
-		for _, ref := range streamRefs {
-			r.NextRange([]*DataRef{ref})
-			_, err := io.Copy(buf, r)
-			require.NoError(t, err)
+	// Setup seed.
+	seed := time.Now().UTC().UnixNano()
+	rand.Seed(seed)
+	msg := seedStr(seed)
+	for _, test := range tests {
```

Comment:
For table based tests like this I generally like to use `t.Run` directives for each one so you get a bit more info about which one failed and can rerun them individually.

Reply:
Sure.